spec/measure_repository_service_test_kit/library_data_requirements_spec.rb

```diff
@@ -0,0 +1,299 @@
+
+      result = run(test, url:, library_id:, library_url:, library_identifier:, library_version:)
+      expect(result.result).to eq('fail')
+    end
+  end
+
+  describe 'Server returns 404 when no library matches id' do
+    let(:test) { group.tests[4] }
+
+    it 'passses when 404 returned with OperationOutcome' do
+      stub_request(
+        :post,
+        "#{url}/Library/INVALID_ID/$data-requirements"
+      ).to_return(status: 404, body: error_outcome.to_json)
+
+      result = run(test, url:)
+      expect(result.result).to eq('pass')
+    end
+
+    it 'fails if 200 status code returned' do
+      stub_request(
+        :post,
+        "#{url}/Library/INVALID_ID/$data-requirements"
+      ).to_return(status: 200, body: error_outcome.to_json)
+
+      result = run(test, url:)
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if library returned' do
+      library = FHIR::Library.new
+      stub_request(
+        :post,
+        "#{url}/Library/INVALID_ID/$data-requirements"
+      ).to_return(status: 200, body: library.to_json)
+      result = run(test, url:)
+      expect(result.result).to eq('fail')
+    end
+  end
+
+  describe 'Server returns 400 when no id, url, or identifier provided' do
+    let(:test) { group.tests[5] }
+
+    it 'passes when 400 returned with OperationOutcome' do
+      stub_request(
+        :post,
+        "#{url}/Library/$data-requirements"
+      ).to_return(status: 400, body: error_outcome.to_json)
+      result = run(test, url:)
+      expect(result.result).to eq('pass')
+    end
+
+    it 'fails if 200 status code returned' do
+      stub_request(
+        :post,
+        "#{url}/Library/$data-requirements"
+      ).to_return(status: 200, body: error_outcome.to_json)
+      result = run(test, url:)
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if library body returned' do
+      library = FHIR::Library.new
+      stub_request(
+        :post,
+        "#{url}/Library/$data-requirements"
+      ).to_return(status: 400, body: library.to_json)
+      result = run(test, url:)
+      expect(result.result).to eq('fail')
+    end
+  end
+
+  describe 'Server returns 400 when id is included in both the path and as a FHIR parameter'
+  let(:test) { group.tests[6] }
+  let(:library_id) { 'library_id' }
+
+  it 'passes when 400 returned with OperationOutcome' do
+    stub_request(
+      :post,
+      "#{url}/Library/#{library_id}/$data-requirements"
+    ).to_return(status: 400, body: error_outcome.to_json)
+    result = run(test, url:, library_id:)
+    expect(result.result).to eq('pass')
+  end
+
+  it 'fails if 200 status code returned' do
+    stub_request(
+      :post,
+      "#{url}/Library/#{library_id}/$data-requirements"
+    ).to_return(status: 200, body: error_outcome.to_json)
+    result = run(test, url:, library_id:)
+    expect(result.result).to eq('fail')
+  end
+
+  it 'fails if library body returned' do
```

Comment:
Same comment as above about the possible addition of a test that checks `fails if server does not return OperationOutcome`